### PR TITLE
WIP: Binary Search Tree and tests

### PR DESCRIPTION
**Pull request by sdweber422**

### CODE REVIEW

File: src/binarySearchTree.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
+'use strict'
+
+class Node {
+
+  constructor( key ) {
+
```
Comment (punitrathore) [inline src/binarySearchTree.js]: its not standard practice to have whitespace between each curly parenthesis.

File: src/binarySearchTree.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    }
+    else {
+      this.insertion( this.root, key )
+    }
+
+  }
+
+  search( key ) {
+
+    this.validateArgs( [...arguments] )
```
Comment (Anasauce) [inline src/binarySearchTree.js:37]: nice condensing logic here

File: src/binarySearchTree.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+  }
+
+  size() {
+
+    return this.count
+
+  }
+
+  // Recursive helper functions
```
Comment (Anasauce) [inline src/binarySearchTree.js:93]: good implementation recursion!

File: src/binarySearchTree.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    while( node.left !== null ) {
+      return this.smallestNodeInRightSubtree( node.left )
+    }
+
+    return node
+  }
+
+  // Error checking
+
+  validateArgs( args ) {
```
Comment (Anasauce) [inline src/binarySearchTree.js:232]: nice, I like how nice and modular your code is...this helper function is a great example!

**Review by Anasauce — COMMENTED** [review]
Great Job!

**Pull request merged**